.github/workflows/on-merge-queue.yml

```diff
@@ -0,0 +1,65 @@
+---
+# Merge queue.  A PR that reaches here has already passed the more rigorous PR
+# suite, so is very likely to pass.  The main reasons for failures here are
+# flaky VMs timing out (which we can't do much about), or a merge conflict
+# with another PR that is also in the merge queue.
+#
+# There's no reason to have multiple stages in this case, because we're
+# expecting it to pass.  Having more than one stage frustrates parallel
+# throughput in low-contention cases, and guarantees a longer critical path.
+name: Merge queue
+on:
+  merge_group:
+    branches: [main, "stable/*"]
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
```

Comment:
This doesn't hurt, but it really shouldn't do anything because we should ever have 2 groups at the same time for commits that have entered the merge queue. If someone pushes a new commit to a enqueued PR it wouldn't be mergeable anymore and a new merge queue job wouldn't start. Lets just leave it though because it doesn't hurt to have this protection in place here too.